src/dos/drives.cpp

```diff
@@ -195,96 +195,174 @@ void DOS_Drive::SetDir(const char *path)
+                                               const std::string& img_name,
+                                               const uint32_t img_size_kb,
+                                               const bool is_hdd)
+{
+	auto image = std::make_unique<imageDisk>(img_file,
+	                                         img_name.c_str(),
+	                                         img_size_kb,
+	                                         is_hdd);
+
+	return numbered_images.emplace_back(std::move(image)).get();
+}
+
+void DriveManager::CloseNumberedImage(const imageDisk* image_ptr)
+{
+	auto matcher = [image_ptr](const std::unique_ptr<imageDisk>& image) {
+		return image.get() == image_ptr;
+	};
+
+	const auto matching_images = std::remove_if(numbered_images.begin(),
+	                                            numbered_images.end(),
+	                                            matcher);
+
+	numbered_images.erase(matching_images, numbered_images.end());
+}
+
+imageDisk* DriveManager::RegisterRawFddImage(FILE* img_file,
+                                             const std::string& img_name,
+                                             const uint32_t img_size_kb)
+{
+	constexpr bool is_hdd = false;
+
+	auto image = std::make_unique<imageDisk>(img_file,
+	                                         img_name.c_str(),
+	                                         img_size_kb,
+	                                         is_hdd);
+
+	return raw_fdd_images.emplace_back(std::move(image)).get();
+}
+
+void DriveManager::CloseRawFddImages()
+{
+	raw_fdd_images.clear();
 }
 
 void DriveManager::InitializeDrive(int drive) {
-	currentDrive = drive;
-	DriveInfo& driveInfo = driveInfos[currentDrive];
-	if (driveInfo.disks.size() > 0) {
+	currentDrive    = check_cast<uint8_t>(drive);
+	auto& driveInfo = drive_infos.at(currentDrive);
+	if (!driveInfo.disks.empty()) {
 		driveInfo.currentDisk = 0;
-		DOS_Drive* disk = driveInfo.disks[driveInfo.currentDisk];
-		Drives[currentDrive] = disk;
-		if (driveInfo.disks.size() > 1) disk->Activate();
+		const auto disk_pointer = driveInfo.disks[driveInfo.currentDisk].get();
+		Drives.at(currentDrive) = disk_pointer;
+		if (disk_pointer && driveInfo.disks.size() > 1) {
+			disk_pointer->Activate();
+		}
 	}
 }
 
 /*
 void DriveManager::CycleDrive(bool pressed) {
-	if (!pressed) return;
-		
-	// do one round through all drives or stop at the next drive with multiple disks
-	int oldDrive = currentDrive;
-	do {
-		currentDrive = (currentDrive + 1) % DOS_DRIVES;
-		int numDisks = driveInfos[currentDrive].disks.size();
-		if (numDisks > 1) break;
-	} while (currentDrive != oldDrive);
+        if (!pressed) return;
+
+        // do one round through all drives or stop at the next drive with
+multiple disks int oldDrive = currentDrive; do { currentDrive = (currentDrive +
+1) % DOS_DRIVES; int numDisks = drive_infos.at(currentDrive).disks.size(); if
+(numDisks > 1) break; } while (currentDrive != oldDrive);
```

Comment:
Maybe keep the formatting, or delete it altogether?

Reply:
Ahh yes - will snip this unused code (can still be found in commit history)